src/dbnode/storage/shard.go

```diff
@@ -2418,6 +2441,21 @@ func (s *dbShard) Snapshot(
 		multiErr = multiErr.Add(err)
 	}
 
+	if multiErr.FinalError() != nil {
```

Comment:
nit: Should this be if final error is nil? Also perhaps let's shortcut and return if err as per normal short-circuiting? i.e. `if err := multiErr.FinalError(); err == nil { return err } ?`